src/edu/csus/ecs/pc2/ui/JListPane.java

```diff
@@ -0,0 +1,149 @@
+		add(scrollPane);
+
+		scrollPane.setViewportView(theList);
+
+		JPanel buttonPane = new JPanel();
+		FlowLayout flowLayout = (FlowLayout) buttonPane.getLayout();
+		flowLayout.setHgap(45);
+		add(buttonPane, BorderLayout.SOUTH);
+
+		JButton saveButton = new JButton("Save");
+		saveButton.addActionListener(new ActionListener() {
+			public void actionPerformed(ActionEvent e) {
+				saveAndClose();
+			}
+		});
+		buttonPane.add(saveButton);
+
+		JButton cancelButton = new JButton("Cancel");
+		cancelButton.addActionListener(new ActionListener() {
+			public void actionPerformed(ActionEvent e) {
+				closeTheWindow();
+			}
+		});
+		buttonPane.add(cancelButton);
+	}
+
+	/**
+	 * Create a pane that has a list of objects.
+	 * 
+	 * 
+	 * @param parentFrame          parent frame for JPanel, will close parent frame
+	 * @param items                list of items to display, must be at least one
+	 *                             item
+	 * @param selectedItemsIndexes selected items indexces
+	 * @param selectedListsSetter  must be not null, invoked if user saves
+	 *                             selections/choices
+	 */
+	@SuppressWarnings("unchecked")
+	public JListPane(JFrame parentFrame, Object [] items, int[] selectedItemsIndexes, ISelectedListsSetter selectedListsSetter) {
+		this();
+		
+		if (selectedListsSetter == null) {
+			throw new IllegalArgumentException("selectedListsSetter is null");
+		}
+		if (parentFrame == null) {
+			throw new IllegalArgumentException("parentFrame is null");
+		}
+
+		if (items == null) {
+			throw new IllegalArgumentException("items is null");
+		}
+
+		if (items.length == 0) {
+			throw new IllegalArgumentException("No items in list");
+		}
+
+		this.parentFrame = parentFrame;
+		theList.removeAll();
+
+		theList.setListData(items);
+
+		this.selectedListsSetter = selectedListsSetter;
+
+		if (selectedItemsIndexes != null) {
+			holdSelectedIndexes = selectedItemsIndexes;
+			theList.setSelectedIndices(selectedItemsIndexes);
+		}
+	}
+
+	@SuppressWarnings("unchecked")
+	protected void closeTheWindow() {
+
+		if (isChanged()) {
+
+			int result = FrameUtilities.yesNoCancelDialog(parentFrame, "Selection changed, save selection?",
+					"Save selection?");
+			if (result == JOptionPane.YES_OPTION) {
+				selectedListsSetter.setSelectedValuesList(theList.getSelectedValuesList(), theList.getSelectedIndices());
+			}
+		}
+
+		parentFrame.setVisible(false);
+		parentFrame.dispose();
+	}
+
+	private boolean isChanged() {
+		int numSelected = theList.getModel().getSize();
+		return numSelected != holdSelectedIndexes.length;
+	}
+
+	@SuppressWarnings("unchecked")
+	protected void saveAndClose() {
+
+		selectedListsSetter.setSelectedValuesList(theList.getSelectedValuesList(), theList.getSelectedIndices());
+		
+		parentFrame.setVisible(false);
+		parentFrame.dispose();
+	}
+
+	@Override
+	public String getPluginTitle() {
+		return "generic JList picker";
+	}
+
+	public int[] getSelectedItemIndex() {
```

Comment:
Since this method can return multiple "selected item indices", I think the method name should be `getSelectedItemIndices()`.   Calling it `getSelectedItemIndex()` implies there can only be a single selected index.

Reply:
will update method name.